DiscountDiscover/View Controllers/ProfileViewController.m

```diff
@@ -26,19 +28,78 @@ @implementation ProfileViewController
-    self.nameLabel.text = user[@"name"];
-    self.usernameLabel.text = [@"@" stringByAppendingString:user.username];
-    self.emailLabel.text = user.email;
-    self.preferencesLabel.text = user[@"preferences"];
-    
-    if (user[@"profileImage"]) {
-        self.profileView = user[@"profileImage"];
+    self.user = [PFUser currentUser];
+    self.nameLabel.text = self.user[@"name"];
+    self.usernameLabel.text = [@"@" stringByAppendingString:self.user.username];
+    self.emailLabel.text = self.user.email;
+    self.preferencesLabel.text = self.user[@"preferences"];
+    
+    if (self.user[@"profileImage"]) {
+        self.profileView.file = self.user[@"profileImage"];
+        [self.profileView loadInBackground];
     }
     self.profileView.layer.cornerRadius = self.profileView.frame.size.width / 2;
     self.profileView.clipsToBounds = YES;
 }
 
+- (IBAction)takePicture:(id)sender {
+    UIImagePickerController *imagePickerVC = [UIImagePickerController new];
+    imagePickerVC.delegate = self;
+    imagePickerVC.allowsEditing = YES;
+    
+    if ([UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera]) {
+        imagePickerVC.sourceType = UIImagePickerControllerSourceTypeCamera;
+    } else {
+        imagePickerVC.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+    }
+
+    [self presentViewController:imagePickerVC animated:YES completion:nil];
+}
+
+- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<NSString *,id> *)info {
+    
+    // get the image captured by the UIImagePickerController
+    UIImage *editedImage = info[UIImagePickerControllerEditedImage];
+    UIImage *resizedImage = [self resizeImage:editedImage withSize: CGSizeMake(500, 500)];
+
+    self.profileView.image = resizedImage;
+    self.user[@"profileImage"] = [ProfileViewController getPFFileFromImage:resizedImage];
+    [self.user saveInBackground];
+    
+    // dismiss UIImagePickerController to go back to your original view controller
+    [self dismissViewControllerAnimated:YES completion:nil];
+}
+
+- (UIImage *)resizeImage:(UIImage *)image withSize:(CGSize)size {
+    
+    UIImageView *resizeImageView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, size.width, size.height)];
+    
+    resizeImageView.contentMode = UIViewContentModeScaleAspectFill;
+    resizeImageView.image = image;
+    
+    UIGraphicsBeginImageContext(size);
+    [resizeImageView.layer renderInContext:UIGraphicsGetCurrentContext()];
+    UIImage *newImage = UIGraphicsGetImageFromCurrentImageContext();
+    UIGraphicsEndImageContext();
+    
+    return newImage;
+}
+
++ (PFFileObject *)getPFFileFromImage: (UIImage * _Nullable)image {
+    
+    if (!image) {
+        return nil;
+    }
+    
+    NSData *imageData = UIImagePNGRepresentation(image);
+    if (!imageData) {
+        return nil;
+    }
+    
+    return [PFFileObject fileObjectWithName:@"image.png" data:imageData];
```

Comment:
Is this always saving a new image to disk? Or overriding the same `image.png` file? Should this file be cleaned after it's saved to the API?